src/Raven.Server/Web/System/CompareExchangeHandler.cs

```diff
@@ -70,5 +71,69 @@ public Task GetCmpXchgValue()
                 }
             }
         }
+        
+        [RavenAction("/databases/*/cmpxchg", "Delete", AuthorizationStatus.ValidUser)]
+        public async Task DeleteCmpXchgValue()
+        {
+            var prefix = Database.Name + "/";
+            var key = prefix + GetStringQueryString("key");
+
+            // ReSharper disable once PossibleInvalidOperationException
+            var index = GetLongQueryString("index", true).Value;
+
+            ServerStore.EnsureNotPassive();
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var command = new RemoveCompareExchangeCommand(key, index);
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    (var raftIndex, var result) = await ServerStore.SendToLeaderAsync(command);
+                    await ServerStore.Cluster.WaitForIndexNotification(raftIndex);
+                    using (context.OpenReadTransaction())
+                    {
+                        var res = ServerStore.Cluster.GetCmpXchg(context, key);
```

Comment:
Why are you doing this? This is invalid.
You are getting this from a totally different value.
You need to use `raftIndex`

Reply:
We try to delete the compare exchange value here, so if we successful (or the value didn't exists at the first place) then res will contain (-1, null).